### PR TITLE
End to end ci

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

**Comment by k-rister** [comment]
@rafaelfolco @atheurer
This is going to need heavy scrutiny and almost certainly cleanup.  There are also some temporary changes in here that will have to be pulled out before final review and merge.

This PR is also dependent on https://github.com/perftool-incubator/rickshaw/pull/184

File: crucible-install.sh
Comment (rafaelfolco) [inline crucible-install.sh]: hacky

File: crucible-install.sh
Comment (rafaelfolco) [inline crucible-install.sh:57]: standalone script downloaded --> install from origin/master

File: crucible-install.sh
Comment (rafaelfolco) [inline crucible-install.sh:61]: upstream is not fetched in these cases:
1) not a fork, user cloned with ssh the main repo --> install from origin/master
2) PR --> enforce --git-repo (or change how the pr checkout works)

File: crucible-install.sh
Comment (rafaelfolco) [inline crucible-install.sh:70]: auto-detection case
you need to fork the main repo to test changes, there is no way to edit a file on the repo cloned with ssh and test it

**Review by rafaelfolco — APPROVED** [review]
lgtm
this looks pretty good !!
perhaps --git-repo opens too many possibilities for problems...
reminder: update INSTALL.md to reflect the new nstaller args...

File: crucible-install.sh
Comment (k-rister) [reply crucible-install.sh]: This gets some history, but it doesn't change anything else about the state of the repository:

https://github.com/perftool-incubator/crucible/runs/3247623465?check_suite_focus=true
https://github.com/k-rister/crucible/runs/3247623305?check_suite_focus=true

File: crucible-install.sh
Comment (k-rister) [reply crucible-install.sh]: I need to do this...

File: crucible-install.sh
Comment (k-rister) [reply crucible-install.sh]: Why is this "hacky"?

File: crucible-install.sh
Comment (k-rister) [reply crucible-install.sh:57]: I should probably add comments to note what the different use cases are that will lead to these different conclusions.

File: crucible-install.sh
Comment (k-rister) [reply crucible-install.sh]: this is done

File: crucible-install.sh
Comment (k-rister) [reply crucible-install.sh]: done

**Review by rafaelfolco — APPROVED** [review]
lgtm

**Pull request merged**
